src/PackageType/BasePackage.php

```diff
@@ -0,0 +1,309 @@
+	 *
+	 * @var string
+	 */
+	protected $author_url = '';
+
+	/**
+	 * Description.
+	 *
+	 * @var string
+	 */
+	protected $description = '';
+
+	/**
+	 * Homepage URL.
+	 *
+	 * @var string
+	 */
+	protected $homepage = '';
+
+	/**
+	 * Whether the package is installed.
+	 *
+	 * @var boolean
+	 */
+	protected $is_installed = false;
+
+	/**
+	 * Releases.
+	 *
+	 * @var Release[]
+	 */
+	protected $releases = [];
+
+	/**
+	 * Package slug.
+	 *
+	 * @var string
+	 */
+	protected $slug = '';
+
+	/**
+	 * Package type.
+	 *
+	 * @var string
+	 */
+	protected $type = '';
+
+	/**
+	 * Magic setter.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @param string $name  Property name.
+	 * @param mixed  $value Property value.
+	 */
+	public function __set( string $name, $value ) {
+		// Don't allow undefined properties to be set.
+		return;
+	}
+
+	/**
+	 * Retrieve the author.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @return string
+	 */
+	public function get_author(): string {
+		return $this->author;
+	}
+
+	/**
+	 * Retrieve the author URL.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @return string
+	 */
+	public function get_author_url(): string {
+		return $this->author_url;
+	}
+
+	/**
+	 * Retrieve the description.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @return string
+	 */
+	public function get_description(): string {
+		return $this->description;
+	}
+
+	/**
+	 * Retrieve the homepage URL.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @return string
+	 */
+	public function get_homepage(): string {
+		return $this->homepage;
+	}
+
+	/**
+	 * Retrieve the name.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @return string
+	 */
+	public function get_name(): string {
+		return $this->name;
+	}
+
+	/**
+	 * Retrieve the slug.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @return string
+	 */
+	public function get_slug(): string {
+		return $this->slug;
+	}
+
+	/**
+	 * Retrieve the package type.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @return string
+	 */
+	public function get_type(): string {
+		return $this->type;
+	}
+
+	/**
+	 * Whether the package is installed.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @return boolean
+	 */
+	public function is_installed(): bool {
+		return $this->is_installed;
+	}
+
+	/**
+	 * Whether the package has any releases.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @return boolean
+	 */
+	public function has_releases(): bool {
+		return ! empty( $this->releases );
+	}
+
+	/**
+	 * Retrieve a release by version.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @param string $version Version string.
+	 * @throws InvalidReleaseVersion If the version is invalid.
+	 * @return Release
+	 */
+	public function get_release( string $version ): Release {
+		if ( ! isset( $this->releases[ $version ] ) ) {
+			throw InvalidReleaseVersion::fromVersion( $version, $this->get_package_name() );
+		}
+
+		return $this->releases[ $version ];
+	}
+
+	/**
+	 * Retrieve releases.
+	 *
+	 * @since 0.3.0
+	 *
+	 * @return Release[]
+	 */
+	public function get_releases(): array {
+		return $this->releases;
+	}
+
+	/**
+	 * Retrieve the latest release version.
```

Comment:
This summary needs clarifying - maybe drop the final word?

Reply:
Good catch. I'll get that fixed up.